recipes/shc_deployer.rb

```diff
@@ -0,0 +1,71 @@
+# coding: UTF-8
+#
+# Cookbook Name:: cerner_splunk
+# Recipe:: shc_deployer
+#
+# Configures the Deployer in a Search Head Cluster
+
+fail 'Deployer installation not currently supported on windows' if platform_family?('windows')
+
+search_heads = CernerSplunk.my_cluster_data(node)['shc_members']
+
+fail 'Search Heads are not configured for sh clustering in the cluster databag' if search_heads.nil? || search_heads.empty?
+
+instance_exec :shc_deployer, &CernerSplunk::NODE_TYPE
+
+include_recipe 'cerner_splunk::_install_server'
+
+execute 'apply-shcluster-bundle' do
+  command lazy { "#{node['splunk']['cmd']} apply shcluster-bundle -target '#{search_heads.first}' --answer-yes -auth admin:#{node.run_state['cerner_splunk']['admin-password']}" }
+  environment 'HOME' => node['splunk']['home']
+  action :nothing
+end
+
+cluster_data = CernerSplunk.my_cluster_data(node) || {}
+
+cluster_bag = CernerSplunk::DataBag.load(cluster_data['apps'], pick_context: ['deployer-apps']) || {}
+
+global_apps_bag = CernerSplunk::DataBag.load(cluster_bag['bag']) || {}
+
+apps = CernerSplunk::SplunkApp.merge_hashes(global_apps_bag, cluster_bag)
+
+# Basic configs for the deployer-configs app
+app_configs = {
```

Comment:
We could have evaluated the contents of the other .conf files here and have created in the app. But proc are not evaluated by the splunk_app resource now. Hence we are writing them to the app later using splunk_template resource.